client/lib/feed-lookup/index.js

```diff
@@ -0,0 +1,52 @@
+// External Dependencies
+var isEmpty = require( 'lodash/isEmpty' ),
+	head = require( 'lodash/head' ),
+	lruCache = require( 'lru-cache' );
+
+// Internal Dependencies
+var inflight = require( 'lib/inflight' ),
+	wpcom = require( 'lib/wp' );
+
+const cache = lruCache( 10 );
+
+function requestKey( feedId ) {
+	return `feed-lookup-${feedId}`;
+}
+
+function discover( feedUrl ) {
+	const key = requestKey( feedUrl );
+
+	if ( inflight.requestInflight( key ) ) {
+		return cache.get( feedUrl );
+	}
+
+	return wpcom.undocumented().discoverFeed(
+		{ url: feedUrl }
+	);
+};
+
+function feedLookup( feedUrl ) {
+	var feedId = cache.get( feedUrl );
```

Comment:
Sure - I was actually wondering what is the most suitable name, but I think your suggestion is fine. See https://github.com/Automattic/wp-calypso/pull/4615/commits/e48d554cd8444fd28d91bcc3a58634c172f7f07f.